[New Rule] Adding Coverage for `AWS IAM or STS API Calls via Temporary Session Tokens`

terrancedejesus opened the pull request:
<!--
Thank you for your interest in and contributing to Detection Rules!
There are a few simple things to check before submitting your pull request
that can help with the review process. You should delete these items
from your submission, but they are here to help bring them to your attention.
-->
# Pull Request

*Issue link(s)*:
* https://github.com/elastic/ia-trade-team/issues/585

<!--
  Add Related Issues / PRs for context. Eg:
    Related to elastic/repo#999
    Resolves #123
  If there is no issue link, take extra care to write a clear summary and label the PR just as you would label an issue to give additional context to reviewers.
-->

## Summary - What I changed
Adds detection coverage for `AWS IAM or STS API Calls via Temporary Session Tokens`. 

Detects use of sensitive AWS STS or IAM API operations using temporary credentials (session tokens starting with 'ASIA'). This may indicate credential theft or abuse of elevated access via a stolen session. It is not common for legitimate users to perform sensitive IAM operations with temporary session tokens.

Additional Information:
* We expect this rule to be tuned over time.
* IAM or STS service API calls with user session tokens are not allowed unless MFA context is in the user session token because of sensitivity
* Typically IAM or STS calls are made with long-term creds, the console, cloud formation or IaC, making this an anomaly
* We focus solely on user identity type being IAMUser as assumed role typically will lead to FPs from service-based workflows requesting temp creds from STS to go about operations

<!--
  Summarize your PR. Animated gifs are 💯. Code snippets are ⚡️. Examples & screenshots are 🔥
-->

## How To Test
* Please review the referenced meta for emulation behavior.
* Query can be used in serverless stack to check for accuracy.

<!--
  Some examples of what you could include here are:
  * Links to GitHub action results for CI test improvements
  * Sample data before/after screenshots (or short videos showing how something works)
  * Copy/pasted commands and output from the testing you did in your local terminal window
  * If tests run in GitHub, you can 🪁or 🔱, respectively, to indicate tests will run in CI
  * Query used in your stack to verify the change
-->

## Checklist

<!-- Delete any items that are not applicable to this PR. -->

- [ ] Added a label for the type of pr: `bug`, `enhancement`, `schema`, `maintenance`, `Rule: New`, `Rule: Deprecation`, `Rule: Tuning`, `Hunt: New`, or `Hunt: Tuning` so guidelines can be generated
- [ ] Added the `meta:rapid-merge` label if planning to merge within 24 hours
- [ ] Secret and sensitive material has been managed correctly
- [ ] Automated testing was updated or added to match the most common scenarios
- [ ] Documentation and comments were added for features that require explanation

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

DefSecSentinel reviewed (approved):
LGTM

imays11 commented:
@terrancedejesus I can't find the emulation details in the Meta you linked. What type of behavior are you hoping to capture here for IAM? From my research into STS abuse I learned that STS short-term tokens created by `GetSessionToken` cannot interact with IAM service (i.e CreateUser), however those created by `AssumeRole` can. Since you're excluding assumedRole identity types then what are you expecting to capture related to IAM? And then for STS, what sort of behavior are you trying to capture?

terrancedejesus commented:
Hey @imays11, thanks for taking a look!

> What type of behavior are you hoping to capture here for IAM?

The goal is to ultimately identify IAM API calls with temporary user session tokens. Which while accepted with MFA context in the session token, would be an interesting signal vs using long-term creds, the console or some IdP.

> From my research into STS abuse I learned that STS short-term tokens created by GetSessionToken cannot interact with IAM service (i.e CreateUser), however those created by AssumeRole can.

Ref: https://docs.aws.amazon.com/STS/latest/APIReference/API_GetSessionToken.html

You can make certain IAM calls with temporary tokens from `GetSessionToken` as long as MFA context exists in the token itself. Typically you would create a virtual device, register it with the IAM user and then enable.

To demonstrate this I...

1. Setup long-term creds locally with one of our AWS IAM users, setting their creds as the default profile for the CLI
2. Created a virtual MFA device - `create-virtual-mfa-device` (linking it to the profile for the CLI)
3. Enabled that device - `enable-mfa-device`
4. Retrieved temporary user session creds 

```
get-session-token \
  --serial-number "$ARN" (ARN of the new MFA device) \
  --token-code "$FINAL_CODE" \
  --duration-seconds 43200 \
  --profile "$AWS_PROFILE" \
  --output json
```

5. Creds were saved as environment variables (AWS CLI prioritizes these first)
6. Launched an IAM API operation script we wrote for our ByBit emulation behavior to check enumerate acceptable vs non-acceptable API calls with the temp creds. (Ping me if you want this.)
7. Checked our stack for CloudTrail logs showcasing this.

<img width="595" alt="Screenshot 2025-04-21 at 5 06 15 PM" src="https://github.com/user-attachments/assets/4d87a1c7-7881-4cee-84f6-a869b2262244" />

<img width="1299" alt="Screenshot 2025-04-21 at 5 10 07 PM" src="https://github.com/user-attachments/assets/3471f5e8-7acf-44f2-a7cd-8717ef9a5476" />


> And then for STS, what sort of behavior are you trying to capture?

I threw STS in there to see what API operations were being called with temporary creds in general. We can remove these and keep it strictly IAM.

imays11 commented:
Thank you for the additional details!
> You can make certain IAM calls with temporary tokens from GetSessionToken as long as MFA context exists in the token itself. Typically you would create a virtual device, register it with the IAM user and then enable. Note I have not tested CRUD operations extensively for this. But I can to be sure, which would influence risk ultimately if allowed or not.

Ahh I missed this MFA exception. I do think we should test which type of operations are actually allowed. From your image it looks like really only "Read" type operations and if that's the case you can probably reduce the risk score. But this would also just be good information to know in general.  I know this was part of your OnWeek project so let me know if you'd like me to test this more extensively as part of BAU AWS work

> Launched an IAM API operation script we wrote for our ByBit emulation behavior to check enumerate acceptable vs non-acceptable API calls with the temp creds. (Ping me if you want this.)

Yes I would love this thank you!

> I threw STS in there to see what API operations were being called with temporary creds in general. We can remove these and keep it strictly IAM.

Yea maybe this rule should stay focused on IAM. In those docs it says _"You cannot call any AWS STS API except AssumeRole or GetCallerIdentity."_  

We have this [Role Chaining](https://github.com/elastic/detection-rules/blob/4ef72457d39e9c065d46aaba0beb052901cad23a/rules/integrations/aws/privilege_escalation_sts_role_chaining.toml#L10) rule to cover AssumeRole and this [Discovery new terms](https://github.com/elastic/detection-rules/blob/4ef72457d39e9c065d46aaba0beb052901cad23a/rules/integrations/aws/discovery_new_terms_sts_getcalleridentity.toml#L12) rule to cover GetCallerIdentity.

Aegrah commented on rules/integrations/aws/persistence_iam_sts_api_calls_via_user_session_token.toml:
```diff
@@ -0,0 +1,87 @@
+language = "kuery"
+license = "Elastic License v2"
+name = "AWS IAM or STS API Calls via Temporary Session Tokens"
+note = """
+## Triage and Analysis
+
+### Investigating AWS IAM or STS API Calls via Temporary Session Tokens
+
+This rule identifies use of temporary AWS credentials (session tokens with prefix `ASIA`) used to perform sensitive IAM API operations.
+These actions may indicate unauthorized privilege escalation or persistence attempts.
+
+#### Possible Investigation Steps
+
+- Identify the IAM user or assumed role and determine if session-based API calls are expected for this user.
+- Review the source IP and user agent for signs of attacker infrastructure (e.g., Kali, VPN).
+- Investigate whether this was preceded by GetSessionToken or credential harvesting activity.
+- Check if the same key or session token was used in other suspicious calls (e.g., CreateAccessKey, PutUserPolicy).
+
+### Response and Remediation
+
+- Invalidate the credentials used and disable the associated IAM user or role if malicious.
+- Rotate secrets and audit permission boundaries.
+- Review activity around the same time and on other AWS services.
+"""
```
Do we want to leverage our Gen-AI tooling to add additional information to this?

terrancedejesus replied:
I'll update this to be more thorough. Thanks Ruben

Aegrah commented on rules/integrations/aws/persistence_iam_sts_api_calls_via_user_session_token.toml:
```diff
@@ -0,0 +1,87 @@
+query = '''
+event.dataset: aws.cloudtrail
+    and event.provider: ("iam.amazonaws.com" or "sts.amazonaws.com")
+    and aws.cloudtrail.user_identity.type:  "IAMUser"
```
```suggestion
    and aws.cloudtrail.user_identity.type: "IAMUser"
```

Aegrah commented on rules/integrations/aws/persistence_iam_sts_api_calls_via_user_session_token.toml:
```diff
@@ -0,0 +1,87 @@
+field = "new_terms_fields"
+value = ["aws.cloudtrail.user_identity.arn"]
+[[rule.new_terms.history_window_start]]
+field = "history_window_start"
+value = "now-14d"
```
```suggestion
field = "history_window_start"
value = "now-14d"

```

terrancedejesus commented:
@imays11 

> From your image it looks like really only "Read" type operations and if that's the case you can probably reduce the risk score.

I agree, reduced to low.

> I do think we should test which type of operations are actually allowed.

I'll test these more extensively this morning.

> Yea maybe this rule should stay focused on IAM. In those docs it says "You cannot call any AWS STS API except AssumeRole or GetCallerIdentity."

Agreed. Removed STS from name and query logic.

I also updated the investigation guide.

imays11 reviewed (approved):
looks good!

terrancedejesus commented:
We have confirmed that with temporary creds you can do CRUD-based API operations on existing assets in AWS.
<img width="1495" alt="Screenshot 2025-04-22 at 11 09 01 AM" src="https://github.com/user-attachments/assets/820cbfbe-a5f4-4de9-9150-84b9c9bdebb4" />

Thanks again @imays11 for the extra attention here!
